### PR TITLE
amp-ad type=taboola

### PR DESCRIPTION
**Pull request by nitzanvolman**
Provides support for amp-ad type=taboola

### CODE REVIEW

**Comment by cramforce** [comment]
Sorry, I did not realize there were overlapping PRs. Made comments on #1518. I believe we should land only this one.

File: ads/taboola.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                countFileds += 1;
+            }
+        }
+        assert(countFileds === 1,
+            'Attribute mismatch. amp-taboola must contain exactly one of those attributes: %s',
+            alternativeFields.join(', '));
+    };
+
+    validateData(data, ['publisher', 'placement', 'mode']);
+    validateExactlyOne(data, ['article', 'video', 'photo', 'search', 'category', 'homepage', 'others']);
```
Comment (jridgewell) [inline ads/taboola.js]: Can we move the above lines outside the `taboola` function? I'm pretty sure they're stateless, which will give marginal benefits when using multiple ads.

**Comment by nitzanvolman** [comment]
@cramforce handled your comments form #1518 squashed and rebased. 
@dvoytenko @Effie - we are getting a validation error on `resizable` attribute on amp-embed and amp-ad. I believe this was valid previously. Is the resizable attribute required for the resizing functionality (the ability for the ad to signal a different size?) 
I will remove the use of it now, so we get a green build.

**Comment by dvoytenko** [comment]
@nitzanvolman if it directly concerns `amp-ad` - no `resizable` attribute is allowed or needed, I believe.

**Comment by nitzanvolman** [comment]
thanks @dvoytenko - so just to clarify, amp will still respond to resize messages from and amp-ad, just the `resize` attribute is redundant? is this correct?

**Comment by dvoytenko** [comment]
@nitzanvolman I believe this is correct. I'll ask @sriramkrish85 to chime in.

**Comment by nitzanvolman** [comment]
Awesome, in that case i think we are ready @dvoytenko - unless anyone sees any issues we missed. 
(@Effie)

**Comment by sriramkrish85** [comment]
@dvoytenko @nitzanvolman is correct -  amp will still respond to resize messages from and amp-ad and does not need a `resizable ` attribute for that. 


https://github.com/ampproject/amphtml/pull/1596

File: 3p/integration.js [hunk 2]
```diff
@@ -35,12 +35,14 @@ import {twitter} from './twitter';
 import {register, run} from '../src/3p';
 import {parseUrl} from '../src/url';
 import {assert} from '../src/asserts';
+import {taboola} from '../ads/taboola';
```
Comment (cramforce) [inline 3p/integration.js]: I added a mechanism to whitelist which embed types are allowed to use `amp-embed`. After rebasing add yourself to `AMP_EMBED_ALLOWED` in this file.

File: ads/taboola.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+             data-publisher="amp-demo"
+             data-mode="thumbnails-a"
+             data-placement="Ads Example"
+             data-article="auto">
+  </amp-embed>
+```
+
+## Configuration
+
+For semantics of configuration, please see ad network documentation.
```
Comment (cramforce) [inline ads/taboola.md]: Can you point to a URL that explains what all these mean?

File: src/3p.js [hunk 4]
```diff
@@ -144,6 +144,41 @@ export function checkData(data, allowedFields) {
 }
 
 /**
+ * Throws an exception if data does not contains a mandatory field.
+ * @param {!Object} data
+ * @param {!Array<string>} mandatoryFields
+ */
+export function validateDataExits(data, mandatoryFields) {
```
Comment (cramforce) [inline src/3p.js]: Typo in Exists.

**Comment by cramforce** [comment]
Re-running tests as that might have been a flake.

Please squash commits and we can merge this. Thanks!

File: ads/taboola.md (hunk 3 above)
Comment (nitzanvolman) [inline ads/taboola.md]: Sure, I will submit that in a separate PR once i get the documentation up.  (I don't want this to delay this merge)

**Comment by nitzanvolman** [comment]
Rebased, squashed.

**Comment by cramforce** [comment]
LGTM and merged. Thanks!

**Pull request merged**